pghoard/pghoard.py

```diff
@@ -419,7 +419,21 @@ def startup_walk_for_missed_files(self):
             # Process uncompressed files (ie WAL pg_receivexlog received)
             for filename in os.listdir(uncompressed_xlog_path):
                 full_path = os.path.join(uncompressed_xlog_path, filename)
-                if not wal.WAL_RE.match(filename) and not wal.TIMELINE_RE.match(filename):
+                if wal.PARTIAL_WAL_RE.match(filename):
+                    # pg_receivewal may have been in the middle of storing WAL file when PGHoard was stopped.
+                    # If the file is 0 or 16 MiB in size it will continue normally but in some cases the file can be
+                    # incomplete causing pg_receivewal to halt processing. Truncating the file to zero bytes correctly
+                    # makes it continue streaming from the beginning of that segment.
+                    file_size = os.stat(full_path).st_size
+                    if file_size in {0, wal.WAL_SEG_SIZE}:
+                        self.log.info("Found partial file %r, size %d bytes", full_path, file_size)
+                    else:
+                        self.log.warning(
+                            "Found partial file %r with unexpected size %d, truncating to zero bytes", full_path, file_size
+                        )
+                        os.truncate(full_path, 0)
```

Comment:
Instead of truncating, how about moving the small file out of way on local disk as it may contain some data that may not exist anywhere else anymore just in case it's needed. Also we'd probably want to increment statsd stats for this so it can be monitored if this happens.

Reply:
I can make the change but I wonder if it is actually possible for the data not to be available. Seems unlikely pg_receivewal would've updated its stream position on PG side before having fully received the segment.

Reply:
Added the copy and stats ops